Homework5/Vacations/Vacation.cs

```diff
@@ -0,0 +1,130 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Vacations
+{
+    internal class Vacation
+    {
+        public List<Employee> SetOfEmployees { get; set; }
+        public Vacation()
+        {
+            SetOfEmployees = new List<Employee>();
+        }
+
+        public void Add(Employee employee)
+        {
+            SetOfEmployees.Add(employee);
+        }
+
+        public double GetAverageDaysOfVacations()
+        {
+            return SetOfEmployees.Average(x => (x.LastDay - x.FirstDay).Days);
+        }
+
+        public List<(string name, int averageDays)> GetAveregeDaysOfVacationsEach()
+        {
+            List<(string, int)> daysEach = new List<(string, int)>();
+            string? name = null;
+
+            foreach (var item in SetOfEmployees)
+            {
+                if (item.Name != name)
+                {
+                    TimeSpan daysSpan = new TimeSpan(0, 0, 0);
+                    int days = 0;
+
+                    foreach (var emp in SetOfEmployees.Where(n => n.Name == item.Name))
+                    {
+                        daysSpan = emp.LastDay - emp.FirstDay;
+                        days = days + daysSpan.Days;
+
+                    }
+                    daysEach.Add((item.Name, days / SetOfEmployees.Where(n => n.Name == item.Name).Count()));
+                }
+                name = item.Name;
+            }
+            return daysEach.DistinctBy(x => x.Item1).ToList();
+        }
+
+        public List<(int month, int empCount)> GetNumberEachMonth()
+        {
+            List<(int, int)> monthsAndNumber = new List<(int, int)>();
+            int count = 0;
+            for (int i = 1; i <= 12; i++)
+            {
+                count = SetOfEmployees.Where(x => x.FirstDay.Month == i || x.LastDay.Month == i).DistinctBy(e => e.Name).ToList().Count();
```

Comment:
When employee has a vacation from month 1 to month 3, current code will miss to include month 2